### DISCUSSION TITLE
How to use DisplayUsing for Name module in details title and avatar menu section.

### DISCUSSION THREAD

**Discussion by beshoo** [laravel/nova-issues#5656] [Q&A] [answered]
- Laravel Version: v10.x
- Nova Version: 4.26.1 (Silver Surfer) 
- PHP Version: 8.x
- Database Driver & Version: Mysql 8.x
- Operating System and Version: Linux
- Browser type and version: Chrom.Latest

### Description:

I am using third-party plugin https://github.com/outl1ne/nova-translatable
This plugin uses to insert the value in the database as a json.

Well, they use DisplayUsing under the hood.
Well, all works just fine but when we try to translate Name of the user module. Nova interface for "Avatar place" and details title are shown as JSON row.

The question is, what is the hook that we can use in the resource to fix that glitch?

![image](https://github.com/laravel/nova-issues/assets/4635115/6bb69381-f760-48c9-abef-ed680fafedb6)

**Comment by beshoo** [comment]
I fixed the name in the User menu but the following workaround

//NovaServiceProvider
```

 public function boot()
    {
        parent::boot();
    
        Nova::userMenu(function (Request $request) {
            if (Str::isJson($request->user()?->name))
            {
                $array = json_decode($request->user()->name, true);
                $request->user()->name = reset($array);
            }
        });

    }
```

How to fix the details section?

**Reply by beshoo** [reply]
It seems there are more places that we can not control:
Details section
Breadcrumbs section

![image](https://github.com/laravel/nova-issues/assets/4635115/4b00548d-6f0d-485c-b4e2-dee743dd0f15)

I hope to have an answer how can we control the visual data to fix such problems on boot via  NovaServiceProvider

**Comment by jigar-dhulla** [comment, marked as answer]
`outl1ne/nova-translatable` uses `spatie/laravel-translatable` make sure it is configured properly, **correct casts and traits are present in the Model**. We are using the same package and Nova seems to be working fine for us.

If configured correctly, `$user->name` should return a string of the current app locale.
